openslides_backend/action/generics.py

```diff
@@ -44,34 +44,26 @@ def create_action_prepare_dataset(self, payload: ActionPayload) -> DataSet:
             relation_fields = []
             additional_instance_fields: Dict[str, List[str]] = defaultdict(list)
             for field_name, field in self.model.get_relation_fields():
-                for instance_field in instance.keys():
-                    if field_name == instance_field:
-                        if field.structured_relation:
-                            if instance.get(field.structured_relation[0]) is None:
-                                raise ActionException(
-                                    "You must give both a relation field "
-                                    "with structured_relation and its corresponding "
-                                    "foreign key field."
-                                )
-                        relation_fields.append((field_name, field))
-                    elif isinstance(field, BaseTemplateRelationField):
-                        regex = (
-                            r"^"
-                            + field_name[: field.index]
-                            + r"\$"
-                            + r"(\d+)"
-                            + field_name[field.index :]
-                            + r"$"
-                        )
-                        match = re.match(regex, instance_field)
-                        if not match:
-                            continue
+                if field_name in instance:
+                    if field.structured_relation:
+                        if instance.get(field.structured_relation[0]) is None:
+                            raise ActionException(
+                                "You must give both a relation field "
+                                "with structured_relation and its corresponding "
+                                "foreign key field."
+                            )
+                    relation_fields.append((field_name, field))
+                elif isinstance(field, BaseTemplateField):
```

Comment:
THis should be BaseTemplateRelationField. Other template fields like amendmend_paragraph are not handled here.

Reply:
Why aren't they handled here? Shouldn't they be working the same way?

Reply:
In line 46 we call get_relation_fields. This should not return non relation fields.

Reply:
Ah, I see. Okay, I'll look into it how to handle this then when looking at #226